### PR TITLE
Add spam report and spam score to email

### PR DESCRIPTION
**Pull request by Uelb**

### CODE REVIEW

File: lib/griddler/email.rb [hunk 1]
```diff
@@ -26,6 +26,10 @@ def initialize(params)
       @raw_headers = params[:headers]
 
       @attachments = params[:attachments]
+
+      @spam_score = extract_spam_score(params[:spam_report])
```
Comment (wingrunr21) [inline lib/griddler/email.rb]: General feedback here: the `@spam_report` assignment could be done before this extract line. Then you could avoid shadowing the `spam_report` reader in the `extract_spam_score` method.

**Comment by wingrunr21** [comment]
So, is the idea here to have the adapters extract their spam reports and send them in as a top-level attribute? Is that preferential vs having the adapters load that via a vendor-specific hash like that implemented in #285?

**Comment by Uelb** [comment]
I think it depends on the state of the art of the different email providers, mandrill does have the spam report attribute but I don't know for the others. If it's not present, the specific hash seems to be the way to go, if it's an information given by all providers, then I think it should be a top level attribute.

**Comment by Uelb** [comment]
Mandrill does it : https://mandrill.zendesk.com/hc/en-us/articles/205583207-What-is-the-format-of-inbound-email-webhooks-
Apparently sengrid sends it too : https://sendgrid.com/docs/Classroom/Basics/Inbound_Parse_Webhook/setting_up_the_inbound_parse_webhook.html#-Default-Parameters
Cloudmailin also : http://docs.cloudmailin.com/features/spam_filtering/

**Comment by wingrunr21** [comment]
Mailgun and Postmark also appear to do it. Ok, then I think we can keep this top level.

**Comment by Uelb** [comment]
Do you want me to rebase my commits ?

File: lib/griddler/email.rb [hunk 2]
```diff
@@ -86,6 +92,10 @@ def extract_cc_from_headers(headers)
       EmailParser.extract_cc(headers)
     end
 
+    def extract_spam_score
+      @spam_report['score'] || @spam_report[:score] if spam_report
```
Comment (wingrunr21) [inline lib/griddler/email.rb]: I think the adapters should be normalizing this attribute. We don't need to support a string and a symbol key. The symbol will work fine.

**Comment by wingrunr21** [comment]
Haha, you are making changes faster than I can comment. One sec!

File: lib/griddler/email.rb [hunk 3]
```diff
@@ -26,6 +26,10 @@ def initialize(params)
       @raw_headers = params[:headers]
 
       @attachments = params[:attachments]
+
+      @spam_report = params[:spam_report]
+
+      @spam_score = extract_spam_score
```
Comment (wingrunr21) [inline lib/griddler/email.rb]: Is memoizing this needed? `extract_spam_score` could be renamed `spam_score`, remove the `attr_reader` and ivar, and just read from the `@spam_report` hash.

**Comment by wingrunr21** [comment]
Ok think I'm good. Sorry about all the comments, just having my coffee now. Just one more suggestion and I think it'll be good.

**Comment by wingrunr21** [comment]
hey, mind rebasing against master and then I'll get this merged. Thanks!

**Comment by wingrunr21** [comment]
@Uelb Github is still reporting conflicts so I can't perform a clean merge

**Comment by Uelb** [comment]
It should be fine now

**Comment by Uelb** [comment]
Wow, just saw hound, I'll fix that...

**Pull request merged**